core/include/traccc/fitting/status_codes.hpp

```diff
@@ -0,0 +1,20 @@
+/** TRACCC library, part of the ACTS project (R&D line)
+ *
+ * (c) 2022-2024 CERN for the benefit of the ACTS project
+ *
+ * Mozilla Public License Version 2.0
+ */
+
+#pragma once
+
+#include <cstdint>
+namespace traccc {
+enum class kalman_fitter_status : uint32_t {
```

Comment:
I'm not in love with the name, but don't have fundamentally better suggestions for the moment. 🤔

But I wonder... Should we rather use `std::uint32_t`? I **think** the C\+\+ standard only guarantees the existence of the `std::` prefixed version.

https://en.cppreference.com/w/cpp/header/cstdint

But we certainly do this pretty inconsistently in the code elsewhere as well. 🤔